src/dialogs/chart_options/chart_options_view.pas

```diff
@@ -0,0 +1,301 @@
+  EditText.Tag := FOOTNOTE_TAG;
+  EditText.OnChange := @SetText;
+  FText[FOOTNOTE_TAG] := EditText;
+  FLabel[FOOTNOTE_TAG] := LabelText;
+  PrevEditText := EditText;
+
+  LabelText := TLabel.Create(TheOwner);
+  LabelText.Parent := self;
+  LabelText.Caption := 'X-axis Title';
+  LabelText.AnchorParallel(akLeft,  10, Self);
+  LabelText.AnchorToNeighbour(akTop, 10, PrevEditText);
+
+  EditText := TCustomEdit.Create(TheOwner);
+  EditText.Parent := self;
+  EditText.AnchorParallel(akLeft, entryBox, Self);
+  EditText.AnchorToNeighbour(akTop, 10, PrevEditText);
+  EditText.AnchorParallel(akRight, 10, Self);
+  EditText.Tag := XTITLE_TAG;
+  EditText.OnChange := @SetText;
+  FText[XTITLE_TAG] := EditText;
+  FLabel[XTITLE_TAG] := LabelText;
+  PrevEditText := EditText;
+
+  LabelText := TLabel.Create(TheOwner);
+  LabelText.Parent := self;
+  LabelText.Caption := 'Y-axis Title';
+  LabelText.AnchorParallel(akLeft,  10, Self);
+  LabelText.AnchorToNeighbour(akTop, 10, PrevEditText);
+
+  EditText := TCustomEdit.Create(TheOwner);
+  EditText.Parent := self;
+  EditText.AnchorParallel(akLeft, entryBox, Self);
+  EditText.AnchorToNeighbour(akTop, 10, PrevEditText);
+  EditText.AnchorParallel(akRight, 10, Self);
+  EditText.Tag := YTITLE_TAG;
+  EditText.OnChange := @SetText;
+  FText[YTITLE_TAG] := EditText;
+  FLabel[YTITLE_TAG] := LabelText;
+  PrevEditText := EditText;
+
+  LabelText := TLabel.Create(TheOwner);
+  LabelText.Parent := self;
+  LabelText.Caption := 'Color selection';
+  LabelText.AnchorParallel(akLeft,  10, Self);
+  LabelText.AnchorToNeighbour(akTop, 10, PrevEditText);
+
+  EditText := TCustomEdit.Create(TheOwner);
+  EditText.Parent := self;
+  EditText.AnchorParallel(akLeft, entryBox, Self);
+  EditText.AnchorToNeighbour(akTop, 10, PrevEditText);
+  EditText.AnchorParallel(akRight, 10, Self);
+  EditText.Tag := COLOR_TAG;
+  EditText.OnChange := @SetText;
+  FText[COLOR_TAG] := EditText;
+  FLabel[COLOR_TAG] := LabelText;
+  PrevEditText := EditText;
+
+  // TODO:  Show 10 color button controls with the standard colors. User can click
+  //        on a button to change it? Maybe a bit too complicated
+  //        Problem is: once a color is set, you have to use the hex colours for all 10
+  //        For this Dialog, just show the colors; don't make them clickable
+
+  LabelText := TLabel.Create(TheOwner);
+  LabelText.Parent := self;
+  LabelText.Caption := 'for date variables, specify min/max as dd' +
+                    DefaultFormatSettings.DateSeparator + 'mm' +
+                    DefaultFormatSettings.DateSeparator + 'yyyy';
+  LabelText.AnchorParallel(akLeft,  2 * entrybox, Self);
+  LabelText.AnchorToNeighbour(akTop, 10, PrevEditText);
+
+  LabelText := TLabel.Create(TheOwner);
+  LabelText.Parent := self;
+  LabelText.Caption := 'X-axis minumum';
+  LabelText.AnchorParallel(akLeft,  10, Self);
+  LabelText.AnchorToNeighbour(akTop, 10, PrevEditText);
+
+  EditText := TCustomEdit.Create(TheOwner);
```

Comment:
Either the view or the datamodel needs some sort of validate of the min/max fields. Since the input is directly written to the script the validation happens during runtime.
It would be nice if this input validation accounts for the datatype of variables on the axis.

Reply:
I was having trouble getting the variable type from the Variables model; in an ideal world, the EditText component would include validation as I wanted to validate min/max on the options view. It may be easier in the contribution unit, but will require a user feedback box (like the help box).

It's also an issue for the command itself. At the moment, the min/max values are evaluated in the command and ignored if they don't match the variable type. This should be checked before the chart is created, and an error issued. I was hoping to be able to get this validation done in AST, but I cannot see a way to condition the option expression types on the variable type.